Bitcoin Support

hvarona opened the pull request:
Added database support for the new type of seeds, and the new type of coin (Bitcoin) and all coin using BIP44.
Create, export and import master seed with bip39.
Added a tab for the bitcoin, Showing to generate the master seed if there is not one.
Import/export bin file wiht the new master seed
Create/import Bitcoin Accounts

kenCode-de commented:
Nelson will start committing your PR's soon, but not just yet. 
I want to get the current version up to date before we start adding in these new features.

hvarona committed:
- Added accountactivitywatcher with socket.io

hvarona committed:
- Adapt BalanceFragment to work with other type of coin
- Added listener to the balance change in account

hvarona committed:
- Fixed Mnemonic, there was no space between brainkey and master seed

hvarona committed:
- Changed GetTransaction

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Changed GetTransaction

hvarona committed:
- Getter and setter for transaction outputs/inputs

hvarona committed:
- Adjust database

hvarona committed:
- Added balance change from insigthapi

hvarona committed:
- Adapting BalanceFragment to show transactions from other coins different from BITSHARES (still in progress...)

bilthon commented on app/src/main/java/de/bitshares_munich/database/SCWallDatabaseContract.java:
```diff
@@ -49,4 +49,89 @@ private SCWallDatabaseContract(){}
+        public static final String TABLE_NAME = "general_accounts";
+        public static final String COLUMN_ID = "id";
+        public static final String COLUMN_NAME = "account_name"; //this is only to tag this account
+        public static final String COLUMN_TYPE = "coin_type";//bitcoin,litecoin
```
I might suggest renaming this constant to COLUMN_COIN_TYPE to completely clarify its meaning and avoid confusion as there already seems to be another COLUMN_TYPE constant being used in another static class.

bilthon commented on app/src/main/java/de/bitshares_munich/database/SCWallDatabaseContract.java:
```diff
@@ -49,4 +49,89 @@ private SCWallDatabaseContract(){}
+        public static final String TABLE_NAME = "graphene_accounts";
+        public static final String COLUMN_ID = "id";
+        public static final String COLUMN_NAME = "account_name"; //The name of the account
+        public static final String COLUMN_TYPE = "network_type";//Steem,BitShares,PeerPlays,Muse
```
Same thing here, I would call this constant COLUMN_NETWORK_TYPE

hvarona replied:
The graphene part is only for reference, i didn't ifully implement it, still i'll change that name.

bilthon commented on app/src/main/java/de/bitshares_munich/database/SCWallDatabaseContract.java:
```diff
@@ -49,4 +49,89 @@ private SCWallDatabaseContract(){}
+    }
+
+    public static class GeneralTransaction implements BaseColumns {
+        public static final String TABLE_NAME = "bitcoin_transaction";
```
If this is a table for general transactions, then its name should reflect that and not be "bitcoin_transaction".

hvarona commented:
The problem with the precision is that the insight-api uses bitcoin directly, if i use the satoshi precision i have to multiply every amount.

hvarona committed:
- Database name change

hvarona committed:
- Using Retrofit and Gson for parsign
- Changing transactionto satoshi units

hvarona committed:
- Implementing bitcoin balance show in BalanceFragment

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Adding the int casting to the fee satoshi convertion

hvarona committed:
- casting long in satoshis conversion

hvarona committed:
- Added GAP to the address calculatet by the GeneralCoinAccount

hvarona committed:
- Added getter, put and modifier for GeneralCoinAddress, GeneratlTransasction, GIOTx

hvarona committed:
- Added empty constructors

hvarona committed:
- Increment database version

hvarona committed:
- Changing int to long in sequence of vin

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Saving Transaction in the Database

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

# Conflicts:
#	app/src/main/java/de/bitsharesmunich/cryptocoincore/insightapi/GetTransactionByAddress.java

hvarona committed:
- Saving Transaction in the Database

hvarona committed:
Merge remote-tracking branch 'origin/bitcoin_support' into bitcoin_support

hvarona committed:
- Saving Transaction in the Database

hvarona committed:
- Finally, the Bitcoin Balance appears in bitcoin tab

hvarona committed:
- Save dataabse address logic

hvarona committed:
- Now a bitcoin balance appears when there is no internet. Begins in 0 if there are no transactions in the db

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Fix Saving transaction and loading on database

hvarona committed:
Merge remote-tracking branch 'origin/bitcoin_support' into bitcoin_support

hvarona committed:
- Added list of transactions to accounts

hvarona committed:
- Read account transaction from database

hvarona committed:
- Change long to float in oldammount and newammount comparison

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Following Transaction confirms

hvarona committed:
- Added precision to the enum Coin
- Fixed the precision of the transactions list

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Insight Socket.io watcher

hvarona committed:
Merge pull request #3 from kenCode-de/develop

Develop

hvarona committed:
Merge branch 'bitcoin_support' into develop

computationalcore committed:
Partial fix for bugs introduced by commit 65efaa932ee6d0bcda6f97bf03616f1b7024aeaa

computationalcore committed:
Partial fix for bugs introduced by commit 65efaa932ee6d0bcda6f97bf03616f1b7024aeaa

computationalcore committed:
Fix conflicts

hvarona committed:
- Added constraints to GeneralTransaction, Inputs and Outputs on the database
- Added accountWatcher subscription

hvarona committed:
- Added constraints to GeneralTransaction, Inputs and Outputs on the database
- Added accountWatcher subscription

hvarona committed:
- Added constraints to GeneralTransaction, Inputs and Outputs on the database
- Added accountWatcher subscription

hvarona committed:
- Fixed an error with GetTransactionByAddress adding repeated in's or out's into the address

hvarona committed:
- Fixed AccountActivityWatcher
- Added GetTransactionData to follow transaction without the minimun confirmations

hvarona committed:
Merge remote-tracking branch 'origin/bitcoin_support' into bitcoin_support

# Conflicts:
#	app/src/main/java/de/bitsharesmunich/cryptocoincore/insightapi/GetTransactionByAddress.java

computationalcore committed:
Fix transaction scroll problem

bilthon committed:
Adding a limit to the amount of times the API call 'get_relative_account_history' is called

bilthon committed:
Merge branch 'computationalcore-master'

bilthon committed:
Version bump

bilthon committed:
Removed fabric file

Commit:
edited node addresses

if public_nodes.json no longer exists, we will need regions too

hvarona committed:
- Change Balance to get uncofirmed and confirmed balances, also to get the less Confirmations
- Added the date on the balance
- Fix the getTransaction to multiplie the time by 1000, the time in the insight api was in seconds

hvarona committed:
- Added confirmations needed to Coin enumeration
- Added confirmations progress to Transactions History and Bitcoin Balance
- Added new colors for confirmations progress

computationalcore committed:
Fix e-Receipt visibility

hvarona committed:
- Change Balance to get uncofirmed and confirmed balances, also to get the less Confirmations
- Added the date on the balance
- Fix the getTransaction to multiplie the time by 1000, the time in the insight api was in seconds

hvarona committed:
- Fixed concurrent action in table adapter

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Now the balanceFragment shows the confirmed balance

hvarona committed:
- Fixed transaction not commiting in updateGeneralTransaction
- Fixed bitcoin balance showing not confirmed balance in the beginning

hvarona committed:
- Adjusting default color of faittext to color.receivedAmount in updateBalanceItem

hvarona committed:
- Change some function in balancefragment to work different when the coin is not bitshares

hvarona committed:
-- Added lessConfirme to balances

hvarona committed:
- Change transaction database to ahve blockheight, indexes and scripts

hvarona committed:
- Change transaction database to ahve blockheight, indexes and scripts

hvarona committed:
- Change transaction database to ahve blockheight, indexes and scripts

hvarona committed:
- Now the confirmations show when the app starts

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Update transaction on Address

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Added label property to Coins, now BTC is printed instead of BITCOIN

hvarona committed:
- Removing the "bit" word for the label BTC in BITCOIN balance

bilthon committed:
Merge branch 'computationalcore-master'

hvarona committed:
- Send logic

hvarona committed:
- Changed the primary key of CryptoCoin objects to an integer

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Send logic
- Database with int ids

hvarona committed:
- Fixed default value of id in GeneralCoinAddress

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

bilthon committed:
Merge branch 'master' into develop

bilthon committed:
Adding support for web wallet backup files

bilthon committed:
Changing the send form verification procedure for more CPU-friendly one

hvarona committed:
- Change database to have the original output transaction
- Fixed utxo calculation

bilthon committed:
Fixing send functionality broken by the modifications of the previous commit

computationalcore committed:
Transaction logic update (partial)

computationalcore committed:
Merge branch 'develop' of https://github.com/kenCode-de/smartcoins-wallet into develop

computationalcore committed:
Fix UIA send transaction

computationalcore committed:
Removed some unused logs

bilthon committed:
Allowing the user to modify an amount read from the QR code

bilthon committed:
Merge branch 'computationalcore-develop' into develop

bilthon committed:
Version bump

bilthon committed:
UI improvements in the dialog and activity that prompt the brainkey backup

bilthon committed:
Adding a new database table used to store all keys created by the app, regardless of their use

bilthon committed:
Storing also imported keys

bilthon committed:
Merge commit 'f6b23efdd4a661c411624'

bilthon committed:
Fixing crash produced by the lack of confirmation button in the PIN dialog UI

hvarona committed:
- Adding a copy of SendScreen Activity to work with GeneralCoins
- Adding a new donation string to work with GeneralCoins

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Added Validator for crypto coin addresses

hvarona committed:
- Create Qr Code for bitcoin
- Changed Validator to include uri genertor and decoder

hvarona committed:
- Now the send activity works with bitcoin
- Fixed repeated "minus" symbol in the amount on transaction history

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

bilthon committed:
Temporarily disabling the backup asset selection and thus the overdraft protection

bilthon committed:
Checking if the fragment is added in order to avoid an IllegalStateException

bilthon committed:
Avoiding potential memory leak by setting a reference to the current BalanceFragment to null

hvarona committed:
- Read QR code to send
- Memo test

computationalcore committed:
Fix OR TextView visibility

bilthon committed:
Merge branch 'develop'

bilthon committed:
Version bump

computationalcore committed:
WIF import (partial)

hvarona committed:
- Fixed receive page not loading "to" and "amount" when request page is called

hvarona committed:
- Change insight api to use coin get precision
- Fixed precision in balances

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Added contact database

hvarona committed:
- Added contact database

hvarona committed:
- Added contact database

computationalcore committed:
WiF (partial)

hvarona committed:
- Fixed balance display issues

hvarona committed:
- Added contact entity

hvarona committed:
Merge remote-tracking branch 'origin/bitcoin_support' into bitcoin_support

hvarona committed:
- Sending Memo
- Receiving Memo on Transaction (miss db and historical data)

bilthon committed:
Hotfix introduced to solve a bug in the account creation procedure

bilthon committed:
Version bump

bilthon committed:
Merge branch 'master' into develop

computationalcore committed:
WiF (partial)

computationalcore committed:
WiF (partial)

hvarona committed:
- Memo parse hex string fix

hvarona committed:
- Memo parse hex string fix

hvarona committed:
- Memo parse hex string fix

hvarona committed:
- Adding contacts for bitcoin (still in progress...)
- Adding field memo to GeneralTransaction

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Save memo on transaction

hvarona committed:
- Added contact manager to dataabse
- Change GIOTx to GTxIO

bilthon committed:
Fixed import issue between the web wallet and the bin backups created by the app

computationalcore committed:
WiF (Import logic finished)

hvarona committed:
- The contact fragmente now works with bitcoin. Bitcoin contacts can be added, deleted and edited.

hvarona committed:
- Change Insight api classes to adapt to the new server addresses
- Updated the Socket.IO Library

hvarona committed:
Merge remote-tracking branch 'origin/bitcoin_support' into bitcoin_support

hvarona committed:
- The SendScreen now can search for bitcoin contacts

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Created the e-receipt for bitcoin

hvarona committed:
Merge branch 'bitcoin_support' into develop

hvarona committed:
Merge pull request #5 from hvarona/develop

Develop

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into develop

hvarona committed:
-Fix classes using deleted class

computationalcore committed:
Fix WIF crash

computationalcore committed:
Fix conflict

computationalcore committed:
Fix Settings Upgrade Account UI button and reformat SettingActivity code

computationalcore committed:
Translated strings related to WIF UI

computationalcore committed:
Export bin backup for WIF (partial)

hvarona committed:
Merge pull request #8 from hvarona/bitcoin_support

merged bs with develop

computationalcore committed:
Fix WIF UI input gravity

hvarona committed:
- DashCoin Logic

hvarona committed:
- DashCoin Logic

hvarona committed:
-Dash Coin Manager

computationalcore committed:
Export bin backup for WIF

computationalcore committed:
Fix package name

computationalcore committed:
Fix "ghost" account export problem for WIF imported account

computationalcore committed:
Loading dialog for WIF import

computationalcore committed:
Brainkey export error message for WIF imported accounts

computationalcore committed:
Import backup bin logic for WIF only based accounts (partial)

hvarona committed:
- Now the balances of all coins show in one page
- Adding a "new account" button next to Balance title (Still in progress...)

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

# Conflicts:
#	app/src/main/java/de/bitsharesmunich/cryptocoincore/fragments/GeneralCoinBalancesFragment.java

hvarona committed:
- DashCoin Manager

hvarona committed:
Merge remote-tracking branch 'origin/bitcoin_support' into bitcoin_support

# Conflicts:
#	app/src/main/java/de/bitsharesmunich/cryptocoincore/base/CryptoCoinFactory.java
#	app/src/main/java/de/bitsharesmunich/cryptocoincore/dash/DashManager.java

computationalcore committed:
Import backup bin logic for WIF only based accounts (finished)

computationalcore committed:
v1.5.14

computationalcore committed:
Improve import wif based account bin backup

computationalcore committed:
Fixed leaked window exception from ImportBackupActivity and ImportWifActivity

hvarona committed:
- Added CryptoCoin Factory newAccount

hvarona committed:
- Added dash definitions

hvarona committed:
- Now there is a button near balances title to add a new coin account

hvarona committed:
- Added buttons next to every type of coin for receiving and sending transactions

hvarona committed:
- LiteCoin and DogeCoin Support

hvarona committed:
- Now all the transactions appear in one place

hvarona committed:
- Added contingency for updateBalanceItem in case there's no balanceGroup added yet.

hvarona committed:
- Fixed e-receipt not appearing when transactions get clicked

hvarona committed:
- Reverted fixed for updateBalanceItem

hvarona committed:
- Added listener event when some BalanceItem gets cleared

hvarona committed:
- Fixed getting count of childs in llbalances

hvarona committed:
- LiteCoin api update

hvarona committed:
- LiteCoin api update

hvarona committed:
- Changed contacts to work only with SCWallDatabase
- Now the user can add a addresses for every coin to the same contact

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Fee Estimation
- Litecoin Dogecoin generalccounfactory

hvarona committed:
Litecoin dogecoin network parameters

hvarona committed:
- Added Litecoin and Doge to Factory

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Added warning message for every address added in a Contact
- Now every address gets validated too

hvarona committed:
- Now the send screen find Contacts by coin

hvarona committed:
- Now when the user create a new coin account, the account is added immediately to the balances list

hvarona committed:
Merge branch 'develop' of https://github.com/kenCode-de/smartcoins-wallet into develop

# Conflicts:
#	app/src/main/java/de/bitshares_munich/database/SCWallDatabaseContract.java
#	app/src/main/java/de/bitshares_munich/database/SCWallSQLiteOpenHelper.java
#	app/src/main/java/de/bitshares_munich/smartcoinswallet/BrainkeyActivity.java
#	app/src/main/java/de/bitshares_munich/smartcoinswallet/ImportBackupActivity.java
#	app/src/main/java/de/bitshares_munich/smartcoinswallet/SettingActivity.java
#	app/src/main/java/de/bitshares_munich/smartcoinswallet/TabActivity.java
#	app/src/main/java/de/bitshares_munich/utils/BinHelper.java
#	app/src/main/java/de/bitsharesmunich/graphenej/FileBin.java
#	app/src/main/res/layout/activity_copybrainkey.xml
#	app/src/main/res/values/strings.xml

hvarona committed:
Merge remote-tracking branch 'remotes/origin/develop' into bitcoin_support

# Conflicts:
#	app/src/main/java/de/bitshares_munich/fragments/BalancesFragment.java

hvarona committed:
- Fixed code to compile (still in progress...)

hvarona committed:
Merging develop branch

hvarona committed:
- Fixed button Import Wallet doesn't work when importing from brainkey.

hvarona committed:
- Removed duplicated EditText in send_screen.xml

hvarona committed:
Make the estimee fee on background task

hvarona committed:
- Removed the bitcoin balance and bitcoin contacts fragments

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Fixed date order in balance history list
- Fixed color in confirmations in balance history list

hvarona committed:
- Fixed "5 of 6 conf" string under bitcoin account balance remaining after transaction totally confirmed

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
Make the estimee fee on background task

hvarona committed:
dash instantsend proccessing

hvarona committed:
- Added "account", "balance_cache" and "spent" fields in GeneralTransaction table
- Now the transaction history table load the altcoins history in batches
- When the order of the transaction history table change, it gets reloaded with a new query to the database

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
dash instantsend proccessing

hvarona committed:
Merge remote-tracking branch 'origin/bitcoin_support' into bitcoin_support

hvarona committed:
- Changed Txi.java file lowercase

hvarona committed:
case sensitive files

hvarona committed:
- Added icon for "unknown coin"
- Changed the coin spinner to disable the coins that are already used

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Now a contact can't have more than one address per coin

hvarona committed:
- Created the tables for Coins Settings
- Added new views to provide a coin selection interface

hvarona committed:
- Bitcoin precision can be set, but it still needs to change how the numbers shows in balances

hvarona committed:
- Now the user can set the precision for all coins. This precision changes the way the ammounts are showed in the balances fragment.

hvarona committed:
- Now all coins show theirs amounts with the right precision when set by the user

hvarona committed:
- The balance fragment gets refreshed immediately after a setting change

hvarona committed:
-Fixed precision setting effect in Bitshares.

hvarona committed:
- Added "Instant send" Dash Coin Setting. (Still doesn't have any effect in the Send Page)

hvarona committed:
- The sendscreen now have a section for the current coin extra settings. The Dash Instant Send setting appears in this section and its default value is the same set by the user in the Dash Settings.

hvarona committed:
- Adding some comments

hvarona committed:
adapting style, adding comments

hvarona committed:
adapting style, adding comments

hvarona committed:
adapting style, adding comments

hvarona committed:
- Documenting code

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- GeneralCoinBalancesFragment commented

hvarona committed:
adapting style, adding comments

hvarona committed:
Merge remote-tracking branch 'origin/bitcoin_support' into bitcoin_support

hvarona committed:
adapting style, adding comments

hvarona committed:
adapting style, adding comments

hvarona committed:
- Commenting contacts code (still in progress...)

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support

hvarona committed:
- Comments for the code

hvarona committed:
- Comments on the code

hvarona committed:
adapting style, adding comments

hvarona committed:
- Comments on the code

hvarona committed:
- Comments on the code

hvarona committed:
adapting style, adding comments

hvarona committed:
- Commenting code

hvarona committed:
Merge branch 'bitcoin_support' of https://github.com/hvarona/smartcoins-wallet into bitcoin_support
